pkgs/applications/misc/obsidian/default.nix

```diff
@@ -34,23 +34,23 @@ in stdenv.mkDerivation rec {
 
   src = fetchurl {
     url =
-      "https://github.com/obsidianmd/obsidian-releases/releases/download/v${version}/obsidian-${version}.asar.gz";
-    sha256 = "AkPx7X00kEds7B1syXJPSV1+TJlqQ7NnR6w9wSG2BRw=";
+      "https://github.com/obsidianmd/obsidian-releases/releases/download/v${version}/obsidian-${version}.tar.gz";
+    sha256 = "1acy0dny04c8rdxqvsq70aa7vd8rgyjarcrf57mhh26ai5wiw81s";
   };
 
   nativeBuildInputs = [ makeWrapper graphicsmagick ];
 
   unpackPhase = ''
-    gzip -dc $src > obsidian.asar
+    tar -xf $src
   '';
 
   installPhase = ''
     mkdir -p $out/bin
 
     makeWrapper ${electron}/bin/electron $out/bin/obsidian \
-      --add-flags $out/share/electron/obsidian.asar
+      --add-flags $out/share/electron/app.asar
```

Comment:
Can we rename the file to obsidian .asar?

Reply:
It might be that `app.asar` is somehow referencing `obsidian.asar`.

Reply:
One more thing: since we're now extracting multiple files, they might not belong under `$out/share/electron` but to a separate `obsidian` dir.

Reply:
I'm not sure I follow on "belong to a separate `obsidian` dir" -- I'm also new to this electron ecosystem. are you suggesting that we put `app.asar` and `obsidian.asar` in different directories? if so, it kind of goes back to the problem of `app.asar` is referencing `obsidian.asar` doesn't it?

Reply:
I would put all the files in one directory, probably `$out/share/electron/obsidian/`. I'm open to suggestions, tho.

Reply:
is that the convention? the nix path already contain `obsidian`, so it didn't feel intuitive to add another directory with that name.

current path is:
`/nix/store/i9k929pw9v8nm73qxlrcn1ri39al8hpr-obsidian-0.11.0/share/electron/{app,obsidian}.asar`

Reply:
This is true until you use
```
environment.systemPackages = [ obsidian ];
```

then the tree will be merged with all the other software you have installed the same way. Same goes for `home-manager`. So if there's any other app that installs `electron/app.asar`, you won't be able to install `obsidian` and viceversa.

Reply:
ah okay that makes sense. does `$out/share/obsidian/{app,obsidian}.asar` sound good to you?

Reply:
Absolutely 👍

Reply:
awesome, done! thanks for taking the time to review and explain 😄

Reply:
No worries and quite the opposite, thanks for contributing!